.github/workflows/main.yml

```diff
@@ -0,0 +1,129 @@
+# This is a main job that handles tests and builds container images.
+
+name: Test, build and push artifacts
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+    paths-ignore:
+      - "docs/**"
+      - "**.md"
+  pull_request:
+    paths-ignore:
+      - "docs/**"
+      - "**.md"
+
+env:
+  # Even though we can test against multiple versions, this one is considered a target version.
+  TARGET_GOLANG_VERSION: "1.18"
+  GOLANG_PROTOBUF_VERSION: "1.28"
+  MOCKGEN_VERSION: "1.6.0"
+  PROTOC_VERSION: "3.19.4"
+
+jobs:
+  test-multiple-go-versions:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: ["1.18"] # As we are relying on generics, we can't go lower than 1.18.
+      fail-fast: false
+    name: Go ${{ matrix.go }} test
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go }}
+      - name: Setup Golang caches
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-golang-${{ matrix.go }}-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-golang-${{ matrix.go }}-
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+        with:
+          version: ${{ env.PROTOC_VERSION }}
+      - name: install protoc-gen-go
+        run: go install "google.golang.org/protobuf/cmd/protoc-gen-go@v${GOLANG_PROTOBUF_VERSION}" && protoc-gen-go --version
```

Comment:
Optional TODO in this PR or future once.

We can add these as make targets in `Makefile` and update `docs/development/README.md` because everyone will need to install these 2 tools anyhow.

Reply:
I like this!

Reply:
@Olshansk updated Makefile, changed the documentation and Dockerfiles and CI to utilize new make command. Thanks for this idea!